### PR TITLE
serde 0.9

### PR DESCRIPTION
**Pull request by AerialX**
Yet another update, probably yet another version bump.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -596,33 +596,34 @@ mod impl_rustc {
...
         }
     }
 
     impl<T: Float + Deserialize> Deserialize for NotNaN<T> {
-        fn deserialize<D: Deserializer>(d: &mut D) -> Result<Self, D::Error> {
+        fn deserialize<D: Deserializer>(d: D) -> Result<Self, D::Error> {
             T::deserialize(d).and_then(|v| {
                 NotNaN::new(v)
-                    .map_err(|_| <D::Error as Error>::invalid_value("value cannot be NaN"))
+                    .map_err(|_| <D::Error as Error>::invalid_value(Unexpected::Float(f64::NAN), &"NaN"))
```
Comment (dtolnay) [inline src/lib.rs]: Please add a unit test for this error message using [this library](https://serde.rs/unit-testing.html). Currently this message would look like:

> invalid value: floating point \`NaN\`, expected NaN

which is probably not what you intended.

**Comment by dtolnay** [comment]
I made the change in #23. Ping @reem.

File: src/lib.rs (hunk 1 above)
Comment (AerialX) [inline src/lib.rs]: Mm, but testing with string comparison seems brittle and relies on serde's internal error string formatting. Does serde_test come with a Deserializer with an error enum that allows you to determine what kind of error it is and the parameters that were passed to it?

Comment (dtolnay) [inline src/lib.rs]: > Mm, but testing with string comparison seems brittle and relies on serde's internal error string formatting.

Exactly the opposite. Testing with string comparison is the only way to test what the user is going to see. If you had written a test saying the error should be equal to `InvalidValue(Float(f64::NAN), "NaN")` it wouldn't have done us any good right?

> Does serde_test come with a Deserializer with an error enum that allows you to determine what kind of error it is and the parameters that were passed to it?

It comes with a Deserializer and an error enum that allows you to determine what error message would be displayed to the user.

Comment (AerialX) [inline src/lib.rs]: If the test were written as `Error::InvalidValue { unexpected: Float(f64::NAN), expected: "NaN" }` it certainly would've. And named arguments would've prevented the mistake in the first place I suppose! [As I've expressed previously](https://github.com/serde-rs/json/issues/285), I'm just not much of a fan of serde's approach to errors, how they generally just devolve into `Message(String)`.

**Comment by dtolnay** [comment]
And now... #27.

**Pull request merged**